pyccel/codegen/printing/ccode.py

```diff
@@ -937,7 +937,8 @@ def _print_IndexedElement(self, expr):
             inds = [self._cast_to(i, NativeInteger(), 8).format(self._print(i)) for i in inds]
         else:
             raise NotImplementedError(expr)
-        return "%s.%s[get_index(%s, %s)]" % (base_name, dtype, base_name, ", ".join(inds))
+        return "GET_ELEMENT(%s, %s, %s)" % (base_name, dtype, ", ".join(inds))
```

Comment:
I think we should raise an error here if `rank>15`

Reply:
would it make sense if I raised the error in `print_Allocate`?

Reply:
That sounds like a better idea, then you're only warned once per array. But what happens if the array is an argument to a function?